synapse/http/server.py

```diff
@@ -133,6 +142,13 @@ def wrapped_request_handler(self, request):
                         },
                         send_cors=True
                     )
+                finally:
+                    try:
+                        request_metrics.stop(
+                            self.clock, request, self.__class__.__name__
+                        )
+                    except:
```

Comment:
Ewwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwww.

Can we not instead just check if `report_metrics` is true?